### ISSUE TITLE
Support `.m` files

### ISSUE THREAD

**Issue by bsansouci** [wokalski/gobi#13]
So I'm working on bindings to some ios library, so I need to have at least one `.m` file which needs to be compiled with `clang` and with a ton of flags

I have a working version that works with _my_ specific file
```
CFLAGS="-isysroot $SDK -isystem $OCAML_COMPILER/lib/ocaml -DCAML_NAME_SPACE -fobjc-arc -miphoneos-version-min=$IOSMINREV"
ARCHNAME=$PLATFORM$BITS
# This ARCHNAME looking inside a dictionary `arch` doesn't work outside of bash 4
# I don't have bash 4 so we'll need another way to get this to work
CC="$TOOLCHAIN_DIR/clang -arch ${arch[$ARCHNAME]}"
$CC $CFLAGS -c -o ${PRODUCT2}.o bindings.m
```

I think we should do the simplest possible thing that could support one or more C file. One potential way would be to provide a default Makefile that grabs all `.m` files and builds them in `_build/tmp/`. This could also generate a `.a` lib file which we'd detect and pack at the end with the rest of the ocaml code.

**Comment by wokalski** [comment]
It should be possible to drag your .m dependencies to the main Xcode project and then link your library to it. (also see #3 )

According to [this](https://caml.inria.fr/pub/docs/manual-ocaml/native.html) `ocamlopt` passes `.c` files down to `cc`, does it work with `.m`, too?

**Comment by wokalski** [comment]
If `ocamlopt` doesn't work with .m, we can use `clang`. The only twist there is that we should make sure to pass the same flags to `clang` when building .m files as `ocamlopt` [uses](https://github.com/wokalski/gobi/blob/master/config/Makefile#L31-L55).

**Comment by wokalski** [comment]
The best option somebody at the irc pointed out is replacing `.m` with `.c` and then passing `ocamlopt -ccopt '-x ObjC++'`. Very very smart.

**Comment by bsansouci** [comment]
Oh I see, so ocamlopt is smart enough to pass it down to clang? I'll try today if I can find some time

**Comment by wokalski** [comment]
yes, it is. It passes down `.c` files and then you trick `clang` into thinking that `.c` files may contain Objective-C.

**Comment by bsansouci** [comment]
I see so the files have to be named `.c`. As long as we document that then it's fine

**Comment by wokalski** [comment]
We can copy & rename them ourselves
If you invoke `gobi file.ml another.m` it should internally copy `another.m` to `another.c`.
